### DISCUSSION TITLE
How use for real trade ?

### DISCUSSION THREAD

**Discussion by tony140222** [rodrigo-brito/ninjabot#355] [Q&A]
### Discussed in https://github.com/rodrigo-brito/ninjabot/discussions/343

<div type='discussions-op-text'>

<sup>Originally posted by **tony140222** July 22, 2024</sup>
I'm looking to use the bot with real account for testing but don't understand how process to done this help will be welcome thanks you all </div>

**Comment by ramilexe** [comment]
@tony140222 you can check this example here: https://github.com/rodrigo-brito/ninjabot/blob/main/examples/futuremarket/futures.go

**Comment by tony140222** [comment]
Just need to set our api and secret there and bot will run and trades with profits how we can see on backtesting?

**Comment by tony140222** [comment]
@ramilexe I've try all the nothing run only get errors to terminal...don't understand how this will work

**Comment by ramilexe** [comment]
@tony140222 What error did you get?

**Comment by tony140222** [comment]
To many for say one specific... :) I just try to found I way to make a trading bot like octobits do if it possible

**Comment by ramilexe** [comment]
Example works fine, without error details not possible to help you

**Comment by tony140222** [comment]
@ramilexe how can I contact you when I'm at home actually to office and need to run again for see error

**Comment by tony140222** [comment]
![17217702283697497228353499265379](https://github.com/user-attachments/assets/05dd6e75-ac7c-46d6-bdef-f4701a614785)
Only past and put api without launch already turn into error

**Comment by panapol-p** [comment]
Hi, @tony140222 
**Steps to Resolve the Issue:**

1. **Install Golang:**  
   Make sure you have Golang installed on your system. You can follow the instructions provided in the [Golang installation guide](https://go.dev/doc/install).

2. **Sync Dependencies:**  
   Run the following command to sync your dependencies:
   ```bash
   go mod tidy
   ```

**Comment by adamantmm** [comment]
The futures example ramilexe linked is the right starting point. A few things that usually trip people up when moving from backtest to live:

1. Drop `WithBacktest` / paper wallet — live needs a real exchange client with API keys that have spot/futures trade permission (read-only keys fail late and confusingly).
2. Start with one pair and tiny size. Strategies that look fine on paper often oversize on the first live candle if position sizing still assumes the backtest portfolio.
3. When you hit errors, paste the first panic/log line literally — “too many errors” is almost always one root cause (invalid pair format, missing market load, or wrong account type for futures).

If you get a concrete stack/log snippet, people can point to the exact config line.
